x-pack/plugins/triggers_actions_ui/public/application/sections/rule_form/rule_add.test.tsx

```diff
@@ -340,87 +340,6 @@ describe('rule_add', () => {
     });
   });
 
-  it('should NOT allow to save the rule if the consumer is not set', async () => {
```

Comment:
I think that should still be a valid test

Reply:
With the new default consumer functionality in the Rule Form Consumer Selector, the consumer can't be unset.

From the issue:
> The benefit of setting [a default] is to improve user experience. This field is required, and if the user hits the save button without scrolling to set this field, nothing happens.

As per discussion starting [here](https://github.com/elastic/kibana/issues/170497#issuecomment-1821798803), we also want the selector to have a **secondary** default in case the specified default is unavailable. The simplest way to do this is by having the UI element automatically select the first element in the dropdown if it receives a `null` `selectedConsumer`: https://github.com/elastic/kibana/pull/171364/files#diff-3610e524bbb22f9e7ff338f5d26ec772a6535ae624b3d624afdad42643206864R126-R127

```js
    // At initialization, select the first value
    if (!validatedSelectedConsumer) onChange(consumers[0] as RuleCreationValidConsumer);
```

I understand this can be confusing because, unfortunately, the `rule_add` Jest tests are actually deeply dependent on the behavior of all the UI elements that `rule_add` embeds, so this test suite is more like an integration test than a typical tightly-scoped unit test. Might be a good idea to refactor this at some point.

Reply:
yes, you are right!